ui/src/app/kvm/components/KVMHeaderForms/KVMHeaderForms.tsx

```diff
@@ -23,19 +24,20 @@ const getFormComponent = (
   clearHeaderContent: ClearHeaderContent,
   setSearchFilter?: SetSearchFilter
 ) => {
-  switch (headerContent.name) {
-    case KVMHeaderNames.ADD_KVM:
+  switch (headerContent.view) {
+    case KVMHeaderViews.ADD_KVM:
       return <AddKVM clearHeaderContent={clearHeaderContent} />;
-    case KVMHeaderNames.COMPOSE_VM:
+    case KVMHeaderViews.COMPOSE_VM:
       return <ComposeForm clearHeaderContent={clearHeaderContent} />;
-    case KVMHeaderNames.DELETE_KVM:
+    case KVMHeaderViews.DELETE_KVM:
       return <DeleteForm clearHeaderContent={clearHeaderContent} />;
-    case KVMHeaderNames.REFRESH_KVM:
+    case KVMHeaderViews.REFRESH_KVM:
       return <RefreshForm clearHeaderContent={clearHeaderContent} />;
     default:
+      const machineHeaderContent = headerContent as MachineHeaderContent;
```

Comment:
I think we should revisit these type casts at some point so maybe we should file an issue and leave a comment (also explain why this is necessary).

Reply:
Done! https://github.com/canonical-web-and-design/maas-ui/issues/3040